fs/fs_zenfs.cc

```diff
@@ -253,11 +256,65 @@ ZenFS::~ZenFS() {
   zbd_->LogZoneUsage();
   LogFiles();
 
+  if (gc_worker_ != nullptr) {
+    run_gc_worker_ = false;
+    gc_worker_->join();
+  }
+
   meta_log_.reset(nullptr);
   ClearFiles();
   delete zbd_;
 }
 
+void ZenFS::GCWorker() {
+  while (run_gc_worker_) {
+    uint64_t non_free = zbd_->GetUsedSpace() + zbd_->GetReclaimableSpace();
+    uint64_t free = zbd_->GetFreeSpace();
+    uint64_t free_percent = (100 * free) / (free + non_free);
+    ZenFSSnapshot snapshot;
+    ZenFSSnapshotOptions options;
+
+    usleep(1000 * 1000 * 10);
+
+    if (free_percent > GC_START_LEVEL) continue;
+
+    options.zone_ = 1;
+    options.zone_file_ = 1;
+    options.log_garbage_ = 1;
+
+    GetZenFSSnapshot(snapshot, options);
+
+    uint64_t threshold = (100 - GC_SLOPE * (GC_START_LEVEL - free_percent));
+    std::set<uint64_t> migrate_zone_ids;
+    for (const auto& zone : snapshot.zones_) {
+      if (zone.capacity == 0) {
+        uint64_t garbage_percent =
```

Comment:
```suggestion
        uint64_t garbage_percent_approximate =
```

The snapshots are not consistent in the face of concurrency. It sort of bothers me, but I don't think it will break anything. Just be aware that the numbers might not add up.